src/diffusers/pipelines/stable_diffusion/safety_checker.py

```diff
@@ -36,8 +36,16 @@ def forward(self, clip_input, images):
         pooled_output = self.vision_model(clip_input)[1]  # pooled_output
         image_embeds = self.visual_projection(pooled_output)
 
-        special_cos_dist = cosine_distance(image_embeds, self.special_care_embeds).cpu().numpy()
-        cos_dist = cosine_distance(image_embeds, self.concept_embeds).cpu().numpy()
+        special_cos_dist = cosine_distance(image_embeds, self.special_care_embeds).cpu()
+        cos_dist = cosine_distance(image_embeds, self.concept_embeds).cpu()
+
+        # cast to float32 to as numpy does not support bfloat16
+        if image_embeds.dtype == torch.bfloat16:
```

Comment:
I don't think we need an `if` statement here as `.float()` should always work and be correct (e.g. we can't do `fp16` on CPU either and doing `numpy()` will move it to `fp32` anyways